packages/arb-token-bridge-ui/src/components/TransferPanel/CustomFeeTokenApprovalDialog.tsx

```diff
@@ -0,0 +1,154 @@
+import { SafeImage } from '../common/SafeImage'
+import { ExternalLink } from '../common/ExternalLink'
+import { useETHPrice } from '../../hooks/useETHPrice'
+import { useNetworksAndSigners } from '../../hooks/useNetworksAndSigners'
+import { formatAmount, formatUSD } from '../../util/NumberUtils'
+import { getExplorerUrl, isNetwork } from '../../util/networks'
+import { useGasPrice } from '../../hooks/useGasPrice'
+import { approveCustomFeeTokenForInboxEstimateGas } from './CustomFeeTokenUtils'
+import { NativeCurrencyErc20 } from '../../hooks/useNativeCurrency'
+
+export type CustomFeeTokenApprovalDialogProps = UseDialogProps & {
+  customFeeToken: NativeCurrencyErc20
+}
+
+export function CustomFeeTokenApprovalDialog(
+  props: CustomFeeTokenApprovalDialogProps
+) {
+  const { customFeeToken, isOpen } = props
+
+  const { ethToUSD } = useETHPrice()
+
+  const { l1, l2 } = useNetworksAndSigners()
+  const { isMainnet } = isNetwork(l1.network.id)
+
+  const { data: l1Signer } = useSigner({ chainId: l1.network.id })
+  const l1GasPrice = useGasPrice({ provider: l1.provider })
+
+  const [checked, setChecked] = useState(false)
+  const [estimatedGas, setEstimatedGas] = useState<BigNumber>(constants.Zero)
+
+  // Estimated gas fees, denominated in Ether, represented as a floating point number
+  const estimatedGasFees = useMemo(
+    () => parseFloat(utils.formatEther(estimatedGas.mul(l1GasPrice))),
+    [estimatedGas, l1GasPrice]
+  )
+
+  const approvalFeeText = useMemo(() => {
+    const eth = formatAmount(estimatedGasFees, { symbol: 'ETH' })
+    const usd = formatUSD(ethToUSD(estimatedGasFees))
+    return `${eth}${isMainnet ? ` (${usd})` : ''}`
+  }, [estimatedGasFees, ethToUSD, isMainnet])
+
+  useEffect(() => {
+    if (!isOpen) {
+      return
+    }
+
+    async function getEstimatedGas() {
+      if (l1Signer) {
+        setEstimatedGas(
+          await approveCustomFeeTokenForInboxEstimateGas({
+            chainProvider: l2.provider,
+            parentChainSigner: l1Signer
+          })
+        )
+      }
+    }
+
+    getEstimatedGas()
+  }, [isOpen, l2.provider, l1Signer])
+
+  function closeWithReset(confirmed: boolean) {
+    props.onClose(confirmed)
+
+    setChecked(false)
+  }
+
+  if (!customFeeToken) {
+    return null
+  }
+
+  return (
+    <Dialog
+      {...props}
+      onClose={closeWithReset}
+      title="Acknowledge approval fees"
+      actionButtonTitle={`Pay approval fee of ${approvalFeeText}`}
+      actionButtonProps={{ disabled: !checked }}
+    >
+      <div className="flex flex-col space-y-6 md:max-w-[490px]">
+        <div className="flex flex-row items-center space-x-4">
+          <SafeImage
+            src={customFeeToken.logoUrl}
+            alt={`${customFeeToken.name} logo`}
+            className="h-8 w-8 grow-0 rounded-full"
+            fallback={
+              <div className="flex h-12 w-12 items-center justify-center rounded-full bg-ocl-blue text-sm font-medium text-white">
```

Comment:
I can do it! I can make it a bit nicer by using the token symbol instead of ?

Reply:
Maybe we can get a different placeholder instead of ?, like some icon. Using token symbol will be a bit awkward because token symbol is often shown next to the symbol. So we will end up with `{tokenSymbol} {tokenSymbol}` in case of a fallback